src/Service/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -0,0 +1,938 @@
+        /// 2) {EntityName}_NoAutoPK -> No auto-generated primary keys present in schema, used for POST requests where PK is not autogenerated and GET (all).
+        /// 3) {EntityName}_NoPK -> No primary keys present in schema, used for POST requests where PK is autogenerated and GET (all).
+        /// Schema objects can be referenced elsewhere in the OpenAPI document with the intent to reduce document verbosity.
+        /// </summary>
+        /// <returns>Collection of schemas for entities defined in the runtime configuration.</returns>
+        private Dictionary<string, OpenApiSchema> CreateComponentSchemas()
+        {
+            Dictionary<string, OpenApiSchema> schemas = new();
+
+            foreach (KeyValuePair<string, DatabaseObject> entityDbMetadataMap in _metadataProvider.EntityToDatabaseObject)
+            {
+                // Entities which disable their REST endpoint must not be included in
+                // the OpenAPI description document.
+                string entityName = entityDbMetadataMap.Key;
+                DatabaseObject dbObject = entityDbMetadataMap.Value;
+
+                if (_runtimeConfig.Entities.TryGetValue(entityName, out Entity? entity) && entity is not null)
+                {
+                    if (entity.GetRestEnabledOrPathSettings() is bool restEnabled)
+                    {
+                        if (!restEnabled)
+                        {
+                            continue;
+                        }
+                    }
+                }
+
+                SourceDefinition sourceDefinition = _metadataProvider.GetSourceDefinition(entityName);
+                HashSet<string> exposedColumnNames = GetExposedColumnNames(entityName, sourceDefinition.Columns.Keys.ToList());
+                HashSet<string> autoGeneratedPKColumnNames = new();
+
+                // Create component schema for FULL entity with all primary key columns (included auto-generated)
+                // which will typically represent the response body of a request or a stored procedure's request body.
+                schemas.Add(entityName, CreateComponentSchema(entityName, fields: exposedColumnNames));
+
+                if (dbObject.SourceType is not SourceType.StoredProcedure)
+                {
+                    // Create an entity's request body component schema excluding autogenerated primary keys.
+                    // A POST request requires any non-autogenerated primary key references to be in the request body.
+                    foreach (string primaryKeyColumn in sourceDefinition.PrimaryKey)
+                    {
+                        // Non-Autogenerated primary key(s) should appear in the request body.
+                        if (!sourceDefinition.Columns[primaryKeyColumn].IsAutoGenerated)
+                        {
+                            autoGeneratedPKColumnNames.Add(primaryKeyColumn);
```

Comment:
The hashset name `autoGeneratedPKColumnNames` is counter intertutive - looks like its storing those PK that are NOT autogenerated. If thats the intention, the name should be fixed

Reply:
updated